numpy/ma/core.py

```diff
@@ -6684,7 +6684,7 @@ def size(obj, axis=None):
 #####--------------------------------------------------------------------------
 #---- --- Extra functions ---
 #####--------------------------------------------------------------------------
-def where (condition, x=None, y=None):
+def where(condition, *args, **kwargs):
```

Comment:
In current master there is a `np._NoValue` class that might be a good default value for `x` and `y`. The test could then be something like
```
if isinstance(x, _NoValue) and isinstance(y, _NoValue):
    # case 1.
elif not isinstance(x, _NoValue) and not isinstance(y, _NoValue):
    # case 2.
else:
    # error
```